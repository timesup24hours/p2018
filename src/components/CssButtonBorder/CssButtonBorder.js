import React from 'react';
import { withRouter } from 'react-router';

import styles from './index.module.css';

export const CssButtonBorder = ({ history }) => {
  const handleOnClick = e => {
    e.currentTarget.classList.toggle(styles.hover);
    history.push('/dashboard_ui');
  };
  return (
    <div className={styles.container}>
      <div
        data-testid="click"
        href=""
        className={styles.a}
        onClick={handleOnClick}
        data-heap-track-id="ui-page-mortgage-button"
      >
        <span className={styles.span} />
        Mortgage
      </div>
      <div className={styles.desc}>
        click above button to another UI only project
      </div>
    </div>
  );
};

export default withRouter(CssButtonBorder);
