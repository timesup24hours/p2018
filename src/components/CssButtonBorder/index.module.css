.container {
  position: relative;
  display: flex;
  justify-content: center;
  align-items: center;
  height: 100%;
  width: 100%;
}
.a {
  position: absolute;
  /* top: 50%;
  left: 50%;
  transform: translate(-50%, -50%); */
  width: 160px;
  height: 60px;
  text-align: center;
  line-height: 60px;
  text-transform: uppercase;
  color: white;
  font-family: sans-serif;
  text-decoration: none;
  transition: 0.5s;
  font-size: 24px;
  overflow: hidden;
  z-index: 0;
}
.a:hover {
  color: black;
  cursor: pointer;
  font-weight: bold;
}
.a.hover {
  color: black;
  cursor: pointer;
  font-weight: bold;
}
.a:before {
  content: '';
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  background: grey;
  z-index: -1;
  transition: 0.5s;
  box-sizing: border-box;
}
.a:hover:before {
  top: calc(100% - 2px);
  background: black;
}
.a.hover:before {
  top: calc(100% - 2px);
  background: black;
}

.a:after {
  content: '';
  position: absolute;
  top: calc(-100% + 2px);
  border: 1px solid black;
  left: 100%;
  width: 100%;
  height: 100%;
  background: black;
  z-index: -1;
  transition: 0.5s;
  box-sizing: border-box;
}
.a:hover:after {
  left: 0;
  transition-delay: 0.5s;
}
.a.hover:after {
  left: 0;
  transition-delay: 0.5s;
}
.span {
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  border-left: 2px solid black;
  border-right: 2px solid black;
  box-sizing: border-box;
  transform-origin: bottom;
  transform: scaleY(0);
  transition: 0.5s;
}
.a:hover span {
  transform: scaleY(1);
  transition-delay: 1s;
}
.a.hover span {
  transform: scaleY(1);
  transition-delay: 1s;
}
.desc {
  position: absolute;
  bottom: 3%;
  left: 50%;
  transform: translateX(-50%);
}
