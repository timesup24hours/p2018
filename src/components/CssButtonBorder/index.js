// import React from 'react';
import CssButtonBorder from './CssButtonBorder';

export default CssButtonBorder;
