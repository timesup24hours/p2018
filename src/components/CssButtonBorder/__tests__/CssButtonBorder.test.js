import React from 'react';
import { shallow, mount } from 'enzyme';

import { CssButtonBorder } from '../CssButtonBorder';

describe('MyNotes.test', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<CssButtonBorder />);

    expect(component).toMatchSnapshot();
  });

  it('should be rendered Looding component', () => {
    const wrapper = shallow(<CssButtonBorder />);
    expect(wrapper.contains('Mortgage')).toBeTruthy();
  });

  it('should be click the button', () => {
    const push = jest.fn();
    const wrapper = mount(<CssButtonBorder />);
    wrapper.setProps({ history: { push } });

    wrapper.find('[data-testid="click"]').simulate('click');
    expect(push).toBeCalled();
  });
});
