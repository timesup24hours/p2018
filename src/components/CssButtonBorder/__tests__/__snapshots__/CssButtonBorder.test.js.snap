// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MyNotes.test should render correctly in "debug" mode 1`] = `
<div
  className="container"
>
  <div
    className="a"
    data-testid="click"
    href=""
    onClick={[Function]}
  >
    <span
      className="span"
    />
    Mortgage
  </div>
  <div
    className="desc"
  >
    click above button to another UI only project
  </div>
</div>
`;
